code/clue.js

```diff
@@ -51,9 +212,110 @@ const pickKiller = () => {
   document.getElementById(
     'killerName'
   ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerOccupation').innerHTML = `${mystery.killer.occupation}`
+  document.getElementById('killerDescription').innerHTML = `${mystery.killer.description}`
+  document.getElementById('killerImage').src = mystery.killer.image
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
+
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+  document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`;
+  document.getElementById('weaponWeight').innerHTML = `${mystery.weapon.weight}`
+  document.getElementById('weaponImage').src = mystery.weapon.image
+
+
+}
+
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms)
+  document.getElementById('roomName').innerHTML = `${mystery.room}`
+}
+
+document.getElementById('killerCard').onclick = pickKiller
+document.getElementById('weaponCard').onclick = pickWeapon
+document.getElementById('roomCard').onclick = pickRoom
+
+// const guess1 = (name) => {
+//   guess = name
+//   document.getElementById('suspectImage').src = guess.image
+
+// }
+
+
+const reveal = () => {
+
+  if (mystery.killer === '' || mystery.weapon === '' || mystery.room === '') {
+    document.getElementById('mystery').innerHTML = 'Please select the mystery answer!'
+  } else if (guess === '') {
+    document.getElementById('mystery').innerHTML = 'Please make a guess!'
+  } else if (guess === mystery.killer) {
+    document.getElementById('mystery').innerHTML = `Correct! It was ${mystery.killer.firstName} ${mystery.killer.lastName} in the
+${mystery.room} with the ${mystery.weapon.name}.`
+  } else if (guess !== mystery.killer) {
+    document.getElementById('mystery').innerHTML = `Wrong! It was ${mystery.killer.firstName} ${mystery.killer.lastName} in the
+${mystery.room} with the ${mystery.weapon.name}.`
+  }
+}
```

Comment:
The logic is solid!   Sometimes it is easier to make different booleans before such an extensive if statement.
For instance, `isSolved` `hasGuessed` `isCorrect`  are some examples of variables that could exist here to make this a bit more readable.